draft-sheffer-wimse-s2s-protocol.md

```diff
@@ -140,9 +140,50 @@ This document uses "service" and "workload" interchangeably. Otherwise, all term
+
+* Preventing Eavesdropping and Interception with TLS
+
+An attacker observing or intercepting the communication channel can view the token and its proof of possession and attempt to replay it to gain an advantage. In order to prevent this the
+token and proof of possession MUST be sent over a secure, server authenticated TLS connection unless a secure channel is provided by some other mechanisms. Host name validation according
+to section TBD MUST be performed. The WIMSE ID token itself is not usable without a proof of possession.
+
+* Limiting Proof of Possession Lifespan
+
+The proof of possession MUST be time limited. A POPs should only for the time necessary for it to be successfully for the purpose it is needed. This will typically be on the order of minutes.  POPs received outside their validity time MUST be rejected.
+
+* Limiting Proof of Possession Scope
+
+The POP MUST have a limited scope. Limitations on scope include constraining the POP to certain receivers and constraining the POP to a usage or transaction.
+
+* Binding to a Nonce
+
+The POP MAY be bound to a fresh nonce to provide single use properties.  The mechanism for distributing a nonce are outside the scope of this specification.
+
+* Binding to Sender
+
+The POP MAY be bound to a sender such as the client identity of a TLS session or TLS channel binding parameters. The mechanisms for binding are outside the scope of this specification.
+
+## Middle Boxes
+
+In some deployments the WIMSE token and proof of possession may pass through multiple systems. The communication between the systems is over TLS, but the token and POP are available in the clear at each intermediary.  While the intermediary cannot modify the token or the information within the POP they can attempt to capture and replay the token or modify the data not protected by the POP. Mitigations listed in the previous section can be used to provide some protection from middle boxes. Deployments should perform analysis on their situation to determine if it is appropriate to trust and allow traffic to pass through a middle box.
+
+## Privacy Considerations
+
+WIMSE tokens and the proofs of possession may contain private information such as user names or other identities. Care should be taken to prevent the disclosure of this information. The use of TLS helps protect the privacy of WIMSE tokens and proofs of possession.
+
+WIMSE tokens are typically associated with a workload that is not directly associated with a user, however in some deployments the workload may be associated directly to a user. In these cases the WIMSE token can be used to track the user if it is disclosed in clear text.
+
+If the WIMSE ID is used with X509 certificates then the workload identity may be disclosed as part of the TLS handshake.  This can be partially mitigated by using TLS 1.3 to protect the client and serve identities.
```

Comment:
```suggestion
If the WIMSE ID is used with X.509 certificates then the workload identity may be disclosed as part of the TLS handshake.  This can be partially mitigated by using TLS 1.3 to protect the client and server identities.
```

Reply:
I don't think that protection of workload identity falls under the "privacy" title. In the service-to-service case, user identity may exist as "context", and this is mentioned above. In the user-to-service case, this would typically be OAuth rather than WIMSE.

Reply:
The connection I made to privacy is that it may be possible that a workload directly maps to a user in some cases.  In this case the workload identity can be used to track the behavior of a user.  To me this is a corner case, but I thought it was worth mentioning.